cpp/IceStorm/clock/Publisher.cpp

```diff
@@ -148,10 +148,26 @@ main(int argc, char* argv[])
             while (true)
             {
                 auto now = chrono::system_clock::to_time_t(chrono::system_clock::now());
+                std::tm timeInfo;
+
+#ifdef _MSC_VER
```

Comment:
I find this also gets in the way of the demo, can we just keep the previous code with a disabled warning.

Reply:
The issue with the previous code is that `localtime` is not thread safe. Are we sure all of these callbacks are being called serially?